docs/book/src/developer/providers/migrations/v1.4-to-v1.5.md

```diff
@@ -3,7 +3,7 @@
 This document provides an overview over relevant changes between Cluster API v1.4 and v1.5 for
 maintainers of providers and consumers of our Go API.
 
-## Minimum Go version
+## Go version
 
 - The Go version used by Cluster API is Go 1.20.x
```

Comment:
We should also include the removal of the Lazy Restmapper featuregate somewhere here IMO.

Reply:
How is this relevant to provider authors?

Reply:
Sounds reasonable. Done